start_control_ssm.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -e
+
+log() {
+  echo "$*" >&2
+}
+
+enable_hybrid_env_ssm() {
+  # SSM parameters necessary to register with a hybrid activation
+  activation_code=$(fetch_from_json '.["ssm"]["activation-code"]' "${USER_DATA}")
+  activation_id=$(fetch_from_json '.["ssm"]["activation-id"]' "${USER_DATA}")
+  region=$(fetch_from_json '.["ssm"]["region"]' "${USER_DATA}")
```

Comment:
Please make scoped variables, well, scoped!

```suggestion
  local activation_code=$(fetch_from_json '.["ssm"]["activation-code"]' "${USER_DATA}")
  local activation_id=$(fetch_from_json '.["ssm"]["activation-id"]' "${USER_DATA}")
  local region=$(fetch_from_json '.["ssm"]["region"]' "${USER_DATA}")
```

Reply:
According to [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155) this can mask return values. How do you feel about:
```
  local activation_code activation_id region
  activation_code=$(fetch_from_json '.["ssm"]["activation-code"]' "${USER_DATA}")
  activation_id=$(fetch_from_json '.["ssm"]["activation-id"]' "${USER_DATA}")
  region=$(fetch_from_json '.["ssm"]["region"]' "${USER_DATA}")
  ```